src/logic/calculate.js

```diff
@@ -0,0 +1,144 @@
+import operate from './operate';
+
+
+const calculate = (dataObj, button) => {
+
+  if (button === 'AC') {
+    return {
+      total: null,
+      next: null,
+      operation: null
+    };
+  }
+
+  if ([0,1,2,3,4,5,6,7,8,9].includes(parseInt(button))) {
+    // If button is '0' and the dataObj is '0' as well irrespective of the operation just return empty dataObj
+    if (button === '0' && dataObj.next === '0') {
```

Comment:
### Division by zero error handling?

I assume this is what this line of code is for. Well and good that you thought about this. However, I would like to add that instead of simply returning an empty object for every time a user attempts division by zero, the better way is to keep an `error` state and display an error message in our `Display` component whenever division by zero happens.